packages/orm/src/newTestInstance.ts

```diff
@@ -308,8 +336,34 @@ export function maybeNew<T extends Entity>(opts?: ActualFactoryOpts<T>): Factory
   return new MaybeNew<T>((opts || {}) as any) as any;
 }
 
+/**
+ * Similar to `maybeNew` in behaviour/use but with enhancements to support polymorphic fields:
+ * 1) Allows you to specify which entity type to create, if it is found a new one is needed
+ * 2) Allows you to prioritize which existing entities to select
+ *
+ * For example below, we are specifying that an Author should be created if needed (and it's default opts),
+ * and also that the priority order for choosing existing entities is Author, Book, and then Publisher
+ *
+ * ```typescript
+ * export function newFavoriteThing(em: EntityManager, opts: FactoryOpts<FavoriteThing> = {}): New<FavoriteThing> {
+ *   return newTestInstance(em, FavoriteThing, {
+ *     parent: maybeNewPoly<FavoriteThingParent>(Author, {}, Book, Publisher),
+ *     ...opts,
+ *   });
+ * }
+ * ```
+ */
+export function maybeNewPoly<T extends Entity>(
+  cstr: EntityConstructor<T>,
+  opts?: ActualFactoryOpts<T>,
+  ...orExisting: EntityConstructor<T>[]
```

Comment:
thought: I'm not married to this signature, and would be open to suggestions.

In particular, it combines the concepts of `this is what I want to make` and `it is also the default thing to look for first`, which probably makes sense in most cases 🤷 

But also, it would be nice if the opts were forced down to be the actual opts of the constructor you specify.

Reply:
Does throwing in another generic help:

```
export function maybeNewPoly<T extends Entity, F extends T>(
  cstr: EntityConstructor<F>,
  opts?: ActualFactoryOpts<F>,
  ...orExisting: EntityConstructor<T>[]
```

Not sure if TS is going to narrow `opts` to be explicitly the union type from cstr...

Reply:
Just thinking out loud, the "array of constructors on the end" seems a little opaque, wdyt of things like:

```
maybeNewPoly<Parent>({
  order: [Foo, Bar, Zaz],
  ifNew: [Bar, { name: "bar props" }]
]);
```

Or maybe that's too much, and:


```
maybeNewPoly<Parent>(
  Bar, { name: "bar props" },
  { order: [Foo, Bar, Zaz] }
]);
```

Which is the existing signature, but opts-izing the last arg solely for readability purposes...wdyt?

Reply:
Yea, I had fool around with adding extra generic types, but it didn't help and made it a bit annoying since one would have to specify the 2 parameters, it wouldn't infer it.

What about this signature:
```
export function maybeNewPoly<T extends Entity>(
  ifNewCstr: EntityConstructor<T>,
  opts?: { ifNewOpts: ActualFactoryOpts<T>; existingOrderSearch: EntityConstructor<T>[] },
): FactoryEntityOpt<T> {
  // Return a marker that resolveFactoryOpt will look for
  return new MaybeNew<T>((opts?.ifNewOpts || {}) as any, opts?.existingOrderSearch ?? [ifNewCstr]) as any;
}
```

Then you could do things like:
```
maybeNewPoly(Activity); 
// Which would be equivalent to 
maybeNewPoly(Activity, { ifNewOpts: {}, existingSearchOrder: [Activity] });
```

Reply:
Her's the 2 type param version, which at least gives folks the option for better typing 🤷 
![image](https://user-images.githubusercontent.com/24765506/162211702-2f454048-a0a9-4ee2-827e-2a17877073f7.png)

Reply:
I'm kind of liking the 2 generic version now more.

@stephenh, if you agree, I'll update the rest of the PR
![image](https://user-images.githubusercontent.com/24765506/162213133-5ab6be29-9b51-4c2a-a92e-f80c75f14f78.png)

Reply:
Updated, TY for the feedback @stephenh 👍